### PR TITLE
[translation] rename package/namespace

### PR DESCRIPTION
**Pull request by kristapratico**
Resolves https://github.com/Azure/azure-sdk-for-python/issues/17043

This renames our package from azure-ai-documenttranslation to azure-ai-translation-document and places the code in the sdk/translation directory. It also adds a namespace package for azure-ai-translation-nspkg.

### CODE REVIEW

File: sdk/translation/azure-ai-translation-nspkg/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Microsoft Azure SDK for Python
+
+This is the Microsoft Azure ai-translation Services namespace package.
```
Comment (maririos) [inline sdk/translation/azure-ai-translation-nspkg/README.md:3]: ohh I have never noticed this.
Is this common for all Python packages? and more for curiosity, do you mind explaining what the purpose of it is?

Comment (kristapratico) [reply sdk/translation/azure-ai-translation-nspkg/README.md:3]: Anything that supports python 2 and extends the `azure` namespace needs a namespace package. We have one for azure, azure.ai (added with text analytics) and now azure.ai.translation.

Comment (maririos) [reply sdk/translation/azure-ai-translation-nspkg/README.md:3]: Thank you!

**Review by maririos — APPROVED** [review]
renames and structure LGTM

File: shared_requirements.txt [hunk 2]
```diff
@@ -1,4 +1,5 @@
 azure-ai-nspkg
+azure-ai-translation-nspkg
```
Comment (maririos) [inline shared_requirements.txt:2]: out of curiosity, why is this not listed for TA or FR?

**Pull request merged**